### ISSUE TITLE
json rules engine for buy/sell

### ISSUE THREAD

**Issue by wayneconnolly** [anmatika/poloniex-api-gui#3]
Hi Anmatika,

You might benefit from an existing rules engine package to introduce for buy/sell (obviously after the PUSH API is connected to returnTicker. Rules could be triggered automatically when a push is received. Users can write their own rules and people could create pull requests to submit strategies. I certainly would :-)

- [https://www.npmjs.com/package/jsrules](https://www.npmjs.com/package/jsrules) (Last update: 1 year ago but looks like the most feature complete for buy/sell rules and easiest to implement and maintain rules)
- [https://www.npmjs.com/package/rule-reactor](https://www.npmjs.com/package/rule-reactor) (Last update: 2 months ago)
- [https://www.npmjs.com/package/node-rules](https://www.npmjs.com/package/node-rules) (Last update: 11 months ago)
- [https://www.npmjs.com/package/json-rules-engine](https://www.npmjs.com/package/json-rules-engine) (Last update: 2 months ago)

**Comment by anmatika** [comment]
Good idea, definitely useful feature. Will implement this after the PUSH API to returnTicker implementation. Not sure if we need a library or just some JSON config for defining the rules. Anyway, have to go through those libraries and see if they add value.

**Comment by wayneconnolly** [comment]
> Not sure if we need a library or just some JSON config for defining the rules

Yes, I agree. The jsrules allows variables to be passed into the method and the conditionals and propisitions can be chained together. It's a gigantic IFTTT which could be handy! The only result is true, false and null. Seems handy! Let me know what you think when you get up to that stage.
